packages/orchestration/src/exos/chain-hub.js

```diff
@@ -454,6 +486,123 @@ export const makeChainHub = (zone, agoricNames, vowTools) => {
         encoding: /** @type {const} */ ('bech32'),
       });
     },
+    /**
+     * Determine the transfer route for a destination and amount given the
+     * current holding chain.
+     *
+     * XXX consider accepting AmountArg #10449
+     *
+     * @param {ChainAddress} destination
+     * @param {DenomAmount} denomAmount
+     * @param {string} holdingChainName
+     * @param {Pick<ForwardInfo['forward'], 'retries' | 'timeout'>} [forwardOpts]
+     * @returns {TransferRoute | undefined} single hop, multi hop, or nothing
+     * @throws {Error} if unable to determine route
+     */
+    makeTransferRoute(destination, denomAmount, holdingChainName, forwardOpts) {
+      const denomDetail = chainHub.getAsset(denomAmount.denom);
+      if (!chainInfos.has(holdingChainName)) {
+        throw makeError(
+          `chain info not found for holding chain: ${q(holdingChainName)}`,
+        );
+      }
```

Comment:
The usual idiom is:

```suggestion
      chainInfos.has(holdingChainName) || Fail`chain info not found for holding chain: ${q(holdingChainName)}`;
```

Is this one of those places where the type checker doesn't grok?

Reply:
Done. There is one condition (`getAsset` can return undefined) but I casted